hack/generate-authors.sh

```diff
@@ -1,15 +1,19 @@
 #!/usr/bin/env bash
+
 set -e
 
-cd "$(dirname "$(readlink -f "$BASH_SOURCE")")/.."
+SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
```

Comment:
Wondering if there was a specific reason for @tianon to use `readlink` here (some corner-case where (sym)links were involved?). Had a look at the original commit, but it didn't make a mention of this (771ed2391536ad722e32be519039454cc98668e8 (https://github.com/moby/moby/pull/7227))

Reply:
`readlink` is intended to resolve symlinks, but `cd && pwd` do the same. `readlink` is not portable.

That being said, I intend to rework all the shell scripts in `hack/` to DRY them up and reduce the boilerplate/kludges. My intention is to simplify all these `SCRIPTDIR` uses so I can replace them all in one go (and this PR is lifted from a monster of a "this bothers me and I'm fixing it" branch that is 3 months old), but first I plan to make everything pass shellcheck.